rfc/ipni-w3c.md

```diff
@@ -146,67 +145,93 @@ flowchart TD
-    w3s-->>Alice: location claims
+    w3s-->>w3s: create/store claims bundle
+    w3s-->>Alice: claims bundle ID and URL
     deactivate w3s
-    activate Alice
-    Alice->>w3s: ipni/offer (inclucion, locations CIDs)
-    deactivate Alice
+    Alice->>w3s: get blob status
+    activate w3s
+    w3s-->>Alice: ok, claims bundle CID
+    deactivate w3s
+    Alice->>w3s: ipni/offer (claims-CID, opt: mh set))
     activate w3s
-    w3s-->>w3s: fetch & verify CARx-index
+    w3s-->>w3s: fetch & verify claims bundle
     w3s-->>w3s: write advertisement
-    w3s-)ipni: publish head (CID)
+    w3s-)ipni: announce head (CID)
     deactivate w3s
     ipni-->>w3s: fetch advert
     activate ipni
     ipni-->>ipni: index entries
     deactivate ipni
 ```
+- User creates their data blob and inclusion claim, then uploads these to w3s to the pre-signed PUT URL.
+  - The inclusion claim listing sub-blocks within the blob is created by the client whether or not they intend to have their data indexed. This is because these blocks may still be needed to find data within the blob, and to allow indexing to be done at a later point without having the re-read the entire blob.
+  - W3s does not need to immediately verify the correctness of the inclusion claim, but can at any time and then replace the inclusion claim with its own to certify that it was verified.
+- w3s creates the location claim for the blob after verifying the content was uploaded. Multiple location claims are created if the data is partitioned into multiple blobs or if the blob is stored in multiple locations.
+  - The client triggers w3s to create the blob location claim(s) when the client is done uploading the data blob(s) to the URLs issued by w3s. By creating the location claims, w3s is responsible for the location of the data, and should verify the data is present before creating the claims.
+  - After triggering location claims creation following blob upload, the client can then issue a request to index the data. This request can be made once anytime during the data's lifetime.
 
-- User uploads their CAR and CARx-index to w3s to the pre-signed PUT url.
-- w3s creates the location claims for the CAR and CARx-index after fetching and verifying the content was uploaded
-  - What triggers the claims creation, polling storage location?
-
-**Location claims** example
+** Inclusion and Location claims** example
 
 ```json
 {
-  "op": "assert/location",
-  "rsc": "https://web3.storage",
-  "input": {
-    "content" : CID /* CAR CID */, 
-    "location": "https://r2.cf/bag...car"
+  "cmd": "assert/inclusion",
+  "sub": "did:key:alice",
+  "args": {
+    "content" : blob-hash,
+    "parts" [
+      [some-block-mh, 128, 32],
+      [other-block-mh, 160, 30],
+      [bafy...cat, 190, 100]
+    ]
+  }
+}
+```
+```json
+{
+  "cmd": "assert/location",
+  "sub": "did:web:web3.storage",
+  "args": {
+    "content" : blob-hash,
+    "url": https://w3s.link/blob-hash
+    "headers": {}
+    "range": [0,1280]
```

Comment:
I think that a location claim should claim one location, instead of bundling multiple locations of some data. Consider the situation where some of the locations are verified by someone and others are not. With single-location claims it seems easier to search for other matching claims from the verifying issuer, instead of looking in different multi-location claims to find matches... but could see this resulting in opinions either way for differen t use cases. so...

Let's do the simplest thing until there is a clear reason to do something more complex.